pkg/virt-manifest/mapper.go

```diff
@@ -0,0 +1,78 @@
+package virt_manifest
+
+import (
+	"encoding/xml"
+	"fmt"
+	"strings"
+
+	"github.com/jeevatkm/go-model"
+	"github.com/libvirt/libvirt-go"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/logging"
+	"kubevirt.io/kubevirt/pkg/virt-handler/virtwrap"
+	"kubevirt.io/kubevirt/pkg/virt-handler/virtwrap/api"
+)
+
+//FIXME: this function is duplicated in virt-handler/virtwrap/manager.go
+func aggregateMappingErrors(mappingErrs []error) error {
+	log := logging.DefaultLogger()
+	if len(mappingErrs) == 0 {
+		return nil
+	}
+	if len(mappingErrs) == 1 {
+		log.Error().Msgf("Error mapping domain spec: %v", mappingErrs[0])
+		return mappingErrs[0]
+	} else {
+		errorStrings := []string{}
+		for _, err := range mappingErrs {
+			errorStrings = append(errorStrings, err.Error())
+		}
+		log.Error().Msgf("Multiple errors mapping domain spec: %s", strings.Join(errorStrings, ", "))
+		return fmt.Errorf("multiple errors mapping domain spec: %s", strings.Join(errorStrings, ", "))
+	}
+}
+
+// This is a simplified version of the domain creation portion of SyncVM. This is intended primarily
+// for mapping the VM spec without starting a domain.
+func MapVM(con virtwrap.Connection, vm *v1.VM) (*v1.VM, error) {
+	vmCopy := &v1.VM{}
+	model.Copy(vmCopy, vm)
+
+	var wantedSpec api.DomainSpec
+	mappingErrs := model.Copy(&wantedSpec, vm.Spec.Domain)
+	log := logging.DefaultLogger()
+
+	if len(mappingErrs) > 0 {
+		return nil, aggregateMappingErrors(mappingErrs)
+	}
+
+	xmlStr, err := xml.Marshal(&wantedSpec)
+	if err != nil {
+		log.Object(vm).Error().Reason(err).Msg("Generating the domain XML failed.")
+		return nil, err
+	}
+
+	log.Object(vm).Info().V(3).Msg("Domain XML generated.")
+	dom, err := con.DomainDefineXML(string(xmlStr))
```

Comment:
Do you envision this randomly generated name to be sent back with the client request or scrubbed before the response is sent?